sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -4392,6 +4392,15 @@ object SQLConf {
     .booleanConf
     .createWithDefault(false)
 
+  val INFER_PANDAS_DICT_AS_MAP = buildConf("spark.sql.execution.pandas.inferPandasDictAsMap")
```

Comment:
Can we make `spark.sql.pyspark.inferNestedDictAsStruct.enabled` as optional, and reuse that configuration?

Reply:
I'm not sure if we can reuse `spark.sql.pyspark.inferNestedDictAsStruct.enabled` for this case?

Because IIRC `spark.sql.pyspark.inferNestedDictAsStruct.enabled` was defined for inferring "nested" `dict` as `StructType` whereas current case is inferring `dict` (non-nested) as `MapType`, so the purpose of two options sounds a bit different for me.

For example, if someone wants to convert a nested `dict` to a `StructType` (should set an option `True`) but at the same time doesn't want to convert the `dict` to a `MapType` (should set an option `False` or `None`), shouldn't there be two separate options?

Please correct me if I missed something!

Reply:
ah okie.

Reply:
If that's the case, we don't need to make this conf optional.

Reply:
Yeah, let me change to `.createWithDefault(false)`